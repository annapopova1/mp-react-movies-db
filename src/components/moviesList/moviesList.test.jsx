import React from 'react';
import { MOVIES_LIST } from '../../../test/movies.test-data';
import MoviesList from './moviesList';

describe('<MoviesList/>', () => {
  test('should render MoviesList component with items', () => {
    const moviesList = shallow(<MoviesList movies={MOVIES_LIST} />);
    expect(moviesList).toMatchSnapshot();
  });

  test('should render message about empty movies list', () => {
    const moviesList = shallow(<MoviesList />);
    expect(moviesList).toMatchSnapshot();
  });
});
