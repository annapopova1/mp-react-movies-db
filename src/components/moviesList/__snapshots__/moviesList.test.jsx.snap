// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesList/> should render MoviesList component with items 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
      key="1"
    >
      <MovieCard
        movie={
          Movie {
            "budget": 0,
            "genres": Array [
              "genre1",
              "genre2",
            ],
            "id": 1,
            "overview": "This is a wider card with supporting text below as a natural lead-in to additional content.",
            "posterPath": "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22348%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20348%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_162cee39977%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A17pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_162cee39977%22%3E%3Crect%20width%3D%22348%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22116.7265625%22%20y%3D%22120.3%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
            "releaseDate": "2010.01.10",
            "releaseYear": 2010,
            "revenue": 0,
            "runtime": 0,
            "tagline": "tagline",
            "title": "Test title",
            "voteAvg": 1.5,
            "voteCount": 0,
          }
        }
      />
    </div>
    <div
      className="col-md-4"
      key="2"
    >
      <MovieCard
        movie={
          Movie {
            "budget": 0,
            "genres": Array [],
            "id": 2,
            "overview": "This is a wider card with supporting text below as a natural lead-in to additional content.",
            "posterPath": "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22348%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20348%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_162cee39977%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A17pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_162cee39977%22%3E%3Crect%20width%3D%22348%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22116.7265625%22%20y%3D%22120.3%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
            "releaseDate": "2004.10.10",
            "releaseYear": 2004,
            "revenue": 0,
            "runtime": 0,
            "tagline": "tagline",
            "title": "Test title",
            "voteAvg": 4,
            "voteCount": 0,
          }
        }
      />
    </div>
    <div
      className="col-md-4"
      key="3"
    >
      <MovieCard
        movie={
          Movie {
            "budget": 0,
            "genres": Array [
              "genre1",
              "genre2",
            ],
            "id": 3,
            "overview": "This is a wider card with supporting text below as a natural lead-in to additional content.",
            "posterPath": "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22348%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20348%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_162cee39977%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A17pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_162cee39977%22%3E%3Crect%20width%3D%22348%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22116.7265625%22%20y%3D%22120.3%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
            "releaseDate": "2000.10.10",
            "releaseYear": 2000,
            "revenue": 0,
            "runtime": 0,
            "tagline": "tagline",
            "title": "Test title",
            "voteAvg": 1.1,
            "voteCount": 0,
          }
        }
      />
    </div>
    <div
      className="col-md-4"
      key="4"
    >
      <MovieCard
        movie={
          Movie {
            "budget": 0,
            "genres": Array [
              "genre1",
            ],
            "id": 4,
            "overview": "This is a wider card with supporting text below as a natural lead-in to additional content.",
            "posterPath": "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22348%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20348%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_162cee39977%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A17pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_162cee39977%22%3E%3Crect%20width%3D%22348%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22116.7265625%22%20y%3D%22120.3%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
            "releaseDate": "2018.10.10",
            "releaseYear": 2018,
            "revenue": 0,
            "runtime": 0,
            "tagline": "tagline",
            "title": "Test title",
            "voteAvg": 2,
            "voteCount": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<MoviesList/> should render message about empty movies list 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1
      className="m-auto"
    >
      No films found
    </h1>
  </div>
</div>
`;
